test/hotspot/jtreg/testlibrary/jvmti/libJvmtiUtils.cpp

```diff
@@ -52,10 +53,34 @@ JNIEXPORT void JNICALL
 Java_jvmti_JVMTIUtils_stopThread(JNIEnv *jni, jclass cls, jthread thread, jobject exception) {
   jvmtiError err =  jvmti->StopThread(thread, exception);
   if (err == JVMTI_ERROR_THREAD_NOT_ALIVE) {
-    LOG("JVMTI_ERROR_THREAD_NOT_ALIVE happened");
+    LOG("JVMTI_ERROR_THREAD_NOT_ALIVE happened\n");
     return;
   }
   check_jvmti_status(jni, err, "Error during StopThread()");
 }
 
+JNIEXPORT void JNICALL
+Java_jvmti_JVMTIUtils_suspendThread(JNIEnv *jni, jclass cls, jthread thread) {
+  jvmtiError err =  jvmti->SuspendThread(thread);
+  if (err == JVMTI_ERROR_THREAD_NOT_ALIVE) {
+    return;
+  }
+  if (err == JVMTI_ERROR_THREAD_SUSPENDED) {
+    return;
+  }
+  check_jvmti_status(jni, err, "Error during SuspendThread");
+}
+
+JNIEXPORT void JNICALL
+Java_jvmti_JVMTIUtils_resumeThread(JNIEnv *jni, jclass cls, jthread thread) {
+  jvmtiError err =  jvmti->ResumeThread(thread);
+  if (err == JVMTI_ERROR_THREAD_NOT_ALIVE) {
+    return;
+  }
+  if (err == JVMTI_ERROR_THREAD_NOT_SUSPENDED) {
+    return;
+  }
+  check_jvmti_status(jni, err, "Error during ResumeThread");
+}
```

Comment:
Would it better to return the jvmtiError status as an int, so it can be checked at the java level?
In fact, it is not good that the `JVMTI_ERROR_THREAD_NOT_ALIVE`, `JVMTI_ERROR_THREAD_SUSPENDED` and `JVMTI_ERROR_THREAD_NOT_SUSPENDED` are always ignored.

Reply:
I was going to ask something similar. Should the tests ever see these errors? If not, probably an exception should be thrown here. If they are ok in some situations, then the tests should be checking for them.